dataplane/forwarding/fwdbuilder/action.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fwdbuilder contains builders for varius proto types.
+package fwdbuilder
+
+import (
+	"encoding/binary"
+
+	fwdpb "github.com/openconfig/lemming/proto/forwarding"
+)
+
+type actionDescBuilder interface {
+	set(*fwdpb.ActionDesc)
+	atype() fwdpb.ActionType
+}
+
+var (
+	// Compile-time checks that builders implement actionDescBuilder interface.
+	_ actionDescBuilder = &LookupActionBuilder{}
+	_ actionDescBuilder = &TransmitActionBuilder{}
+	_ actionDescBuilder = &LookupActionBuilder{}
+)
+
+// ActionBuilder is a builder for forward action types.
+type ActionBuilder struct {
+	adb actionDescBuilder
+}
+
+// Action returns a new action builder.
+func Action(adb actionDescBuilder) *ActionBuilder {
```

Comment:
I think it's clearer if this is called NewAction or actually removed because protobuf's builder API doesn't have this.

Also consider using the same API style as protobufs where you populate a builder struct -- this looks very similar to Ondatra's style but we may not necessarily need the verbosity of a fluent API.

Reply:
i prefer this approach https://github.com/kubernetes/client-go/blob/master/applyconfigurations/core/v1/pod.go

see next PR for how APU is used, it's a significant improvement

Reply:
Thanks for the link. There the object is called AppliedConfiguration. Do you think it's clearer if the package is called `fwdconfig`?

Reply:
sgtm